File: 004/4.c
// Find the largest palindrome made from the product of two 3-digit numbers.

#include <stdio.h>

int main () {

	int range = 99;

	int a = range;
	int b = range;
	int reset = range;
	int palim; // Mir ist schon klar, dass es nicht paliMdrom heißt^^ ist ein Witz meinerseits 
	int maxPalim;

	while (1) {
		palim = a * b;
		if (isPalim(palim)) {
			if (maxPalim < palim) {
				maxPalim = palim;
				printf("%i\n", maxPalim);
			}
		}

		if (b > 1) {
			b--;
		} else {
			b = reset;
			a--;
		}

		if (a < 1) {
			break;
		}
	}

	return 0;
}

int isPalim (int orig) {
	int rev = 0;
	int zahl = orig;

	while (zahl > 1) {
		rev = rev * 10 + zahl % 10;
		zahl /= 10;
	}

	if (rev == orig) {
		return 1;
	} else {
		return 0;
	}
}

File: 004/4.swift
// A palindromic number reads the same both ways. The largest palindrome made
// from the product of two 2-digit numbers is 9009 = 91 × 99.
//
// Find the largest palindrome made from the product of two 3-digit numbers.

func isPalindrome(number: Int) -> Bool {
    var rev = 0
    var num = number

    while num != 0 {
        rev = rev * 10 + num % 10
        num = num / 10
    }

    if rev == number {
        return true
    }
    return false
}

let range = 999
var palindromes = [Int]()

for a in (1...range).reverse() {
    for b in (1...range).reverse() {
        if isPalindrome(a * b) {
            palindromes.append(a * b)
        }
    }
}

print(palindromes.sort().last!)
